ArduPlane/quadplane.cpp

```diff
@@ -1937,7 +1937,38 @@ void QuadPlane::motors_output(bool run_rate_controller)
         if (now - last_att_control_ms > 100) {
             // relax if have been inactive
             relax_attitude_control();
+            /*
+              when starting the VTOL motors force use of the fixed
+              wing controller for attitude control if we are outside
+              of the Q_ANGLE_MAX limit. We keep it forced until we
+              have recovered to within the limit
+
+              This avoids an issue with the attitude controller
+              limiting the rate of recovery and also an issue with the
+              quaternion controller trying to fix the attitude in a
+              way that is bad for fixed wing aircraft (eg. putting
+              nose down when inverted, leading to rapid loss of
+              height)
+
+              we don't do this in QACRO or tailsitter modes, as this
+              type of attitude is expected
+            */
+            if (!tailsitter.enabled() &&
+                plane.control_mode != &plane.mode_qacro &&
+                (abs(ahrs_view->roll_sensor) > aparm.angle_max ||
+                 abs(ahrs_view->pitch_sensor) > aparm.angle_max)) {
```

Comment:
I think this should use the larger of angle max and the fixedwing angle limits. Most vehicles have a larger roll limit in FW flight than they do in VTOL.

Reply:
I wonder if we could re-use the angle assistance thresholds, the use case is the same really.

https://github.com/ArduPilot/ardupilot/blob/9fb1e2b9f7b746716397aa1d298943c201d9dd8a/ArduPlane/VTOL_Assist.cpp#L121-L133

Don't need the trigger delay of course in this case. We may not even have to check here, we could check if we were in angle assistance when the VTOL mode was entered.

Reply:
My thinking was just that we don't want a change in behavior from what we have now in the case that a user switches modes when the vehicle is within the expected attitude envelope for fixed-wing flight.

Is the goal that this should be a special case that kick in to recover from a unexpected attitude or should it kick in the course of a typical flight?

Reply:
I actually do want a change in behaviour. When the two angle limits are similar it doesn't matter, but when the fixed wing roll limit is a lot larger than the VTOL limit then the vehicle starts flying outside the envelope where it has been tuned and I think it is a reasonable to expect the vehicle to come back within the VTOL angle limit as quickly as possible.